app/src/contexts/surveyContext.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { createContext, PropsWithChildren, useContext, useEffect } from 'react';
+import { useParams } from "react-router";
+
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import useDataLoader from 'hooks/useDataLoader';
+import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
+
+
+
+export interface ISurveyContext {
+  surveyWithDetails: IGetSurveyForViewResponse | null;
+  refresh(): void;
+  _setSurveyId(surveyId: number): void;
+  _setProjectId(projectId: number): void;
+}
+
+const SurveyContext = createContext<ISurveyContext>({
+  surveyWithDetails: null,
+  refresh: () => null,
+  _setSurveyId: () => null,
+  _setProjectId: () => null,
+});
+
+export const SurveyContextProvider = (props: PropsWithChildren<void>) => {
+  const [surveyId, setSurveyId] = React.useState<number | null>(null);
+  const [projectId, setProjectId] = React.useState<number | null>(null);
+
+  const biohubApi = useBiohubApi();
+
+  const surveyDataLoader = useDataLoader((project_id: number, survey_id: number) => {
+    return biohubApi.survey.getSurveyForView(project_id, survey_id);
+  });
+
+  const refresh = () => {
+    if (projectId && surveyId) {
+      surveyDataLoader.refresh(projectId, surveyId)
+    }
+  }
+
+  useEffect(() => {
+    refresh()
+  }, [projectId, surveyId])
+
+  const surveyContext: ISurveyContext = {
+    surveyWithDetails: surveyDataLoader.data || null,
```

Comment:
I wonder if you could just return the dataloader itself, and then the children could reference .data or .error, etc.